### PR TITLE
5661 statistics to badges

### LINKED ISSUE #5661: Transform the achievement page's "statistics" section into a "badges" section
Similar to GitHub, with for instance a `x3` to say that for instance 3 thanks were received:

Screenshot 2024-03-27 at 16 45 44

Tapping each badge would reveal what the badge means.

Badges with zero could be greyed out to generate interest without the negativity of a `0`.

Currently it looks like this:
Screenshot 2024-03-27 at 16 49 38

### PR DESCRIPTION
**Pull request by YujingZHG**
**Description (required)**

Fixes #5661 

What changes did you make and why?

Transformed the statistics of achievement into badges. Each badge changes to gray and hides its count if the value is 0. 

- Added badgeRender(Map<ImageView, TextView> badgeMap) function, used to render badges with a filter, and the GrayFilter(ImageView imageView) function applies the gray filter. 

- Used Google’s FlexboxLayout to enable badges to wrap automatically in the layout.

- Added String Badges in Strings.xml, drawable badge_number_background.xml in drawable for displaying.

**Tests performed (required)**

Tested AchievementsFragmentUnitTests on Medium Phone API 33 (Android 13.0 "Tiramisu" | arm64 emulator).

**Screenshots (for UI changes only)**
Set badges numbers to 0
image
None 0 badges
image

---

- [x] I've read CONTRIBUTING.md

### CODE REVIEW

**Comment by nicolas-raoul** [comment]
Looks great! :-)

**Comment by YujingZHG** [comment]
Hi  @nicolas-raoul what should I do with the failing check? or just leave it there?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I just tested, it is generally great, I only found these issues:

- Tapping the "featured images" icon shows the "thanks received" tooltip.
- Tapping the "quality images" icon shows the "featured images" tooltip.
- I am pretty sure I have 100+ nearby uploads and zero featured images, but I see this:


If the count was already incorrect in the main branch, then no need to fix the count.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -370,7 +370,8 @@
   <string name="delete">Delete</string>
   <string name="Achievements">Achievements</string>
   <string name="Profile">Profile</string>
-  <string name="statistics">Statistics</string>
+  <string name="statistics">Badges</string>
```
Comment (nicolas-raoul) [inline app/src/main/res/values/strings.xml:373]: Can this one be removed?

**Comment by nicolas-raoul** [comment]
I will check whether unit tests fail in master. If yes then don't worry.

**Comment by nicolas-raoul** [comment]
Please do not worry about unit tests, this is tracked at #5921

But do you have time to check https://github.com/commons-app/apps-android-commons/pull/5917#pullrequestreview-2412011916 ? :-)

**Comment by nicolas-raoul** [comment]
@YujingZHG Did you get a chance to check? A small conflict must also be fixed. Is it OK for someone else to continue your work? Thanks! :-)

**Comment by YujingZHG** [comment]
@nicolas-raoul  Yes. Thanks :)

**Comment by nicolas-raoul** [comment]
Thanks all! :-)

**Pull request closed without merging**